### ISSUE TITLE
Handle LSP requests when blocking while flushing writes

### ISSUE THREAD

**Issue by the-mikedavis** [helix-editor/helix#12176]
https://github.com/helix-editor/helix/discussions/11902#discussioncomment-11390451 points out a case where a language server appears to be awaiting a response to `workspace/configuration` in order to respond to `textDocument/formatting`. This is a problem for `:write-quit` since we block the main thread waiting for the `textDocument/formatting` response while the main thread is also responsible for replying to LSP server requests. Ideally we should be able to continue responding to LSP requests while awaiting shutdown.

**Comment by pascalkuthe** [comment]
yeah I suspected that our write flushing would eventually be an issue somewhere. We should not be shutting down the editor until writes are done. Ideally we would even show something that indicates the editor is still saving (and allow cancelling the shutdown ala #6436)
